### ISSUE TITLE
Set Variable If keyword has long runtime in RF6.1

### ISSUE THREAD

**Issue by jeffreyt6422** [robotframework/robotframework#4917]
The attached test took less than 100 ms to run on RF6.0.2, but more than 40 s to run on RF6.1.
Windows 10 (22H2 build 19045.3570), Python 3.11.3.
The time taken to run appears to be an exponential relationship to the value in ${var1} (at least initially) when run in RF6.1.
[setvariableif.txt](https://github.com/robotframework/robotframework/files/13208465/setvariableif.txt)

**Comment by pekkaklarck** [comment]
I cannot reproduce this. I noticed a minimal different between v6.0 and v6.1, but it was ~5% and not orders of magnitudes like 100ms vs 40s. According to git bisect that small performance degradation was caused by a9160a8, but that looks really odd. I believe the difference was just so small that it was hard to get reliable measurements.

Interestingly we recently had another issue about Set Variable If performance (#4886). That was about performance getting worse if there are lot of conditions and only some of the latter ones matches. That error I could reproduce, although I needed much more conditions on my machine, and the issue has been fixed. Good news is that the fix affects also this usage and the provided example was clearly fastest with the current code.

In other words, it is true that Set Variable If performance has been poor if there are lot of conditions like in the provided example. I cannot, however, reproduce performance being considerably worse with v6.1 than it was v6.0. I hope you can investigate the situation on your machine to see is there something else different than the version.

**Comment by pekkaklarck** [comment]
No new info. Closing.
